src/Compilers/Core/Portable/PEWriter/MetadataWriter.PortablePdb.cs

```diff
@@ -888,6 +888,9 @@ private void EmbedCompilationOptions(BlobReader? pdbCompilationOptionsReader, Co
                     WriteValue(CompilationOptionNames.Optimization, optimizationLevel.ToPdbSerializedString(debugPlusMode));
                 }
 
+                var platform = module.CommonCompilation.Options.Platform;
+                WriteValue(CompilationOptionNames.Platform, platform.ToString());
```

Comment:
Platform can't be determined from the CLR/PE header flags?

Reply:
It may be possible, but I wasn't able to get all the values to round trip. The CoffHeader.Machine, for example, is not enough information. There must be other bits that imply `Platform.AnyCpu` and `Platform.AnyCpu32BitPreferred` that I just wasn't able to find.

Reply:
These are stored in CorHeader flags:
https://source.dot.net/#System.Reflection.Metadata/System/Reflection/PortableExecutable/CorHeader.cs,1f6cfc9c2185aaf4,references

Reply:
Filed #51941 to make sure we get to this. Thanks